ManalHamdi/assignment2/Assignment2_q2.cc

```diff
@@ -0,0 +1,101 @@
+#include <iostream>
+#include <queue>
+#include <vector>
+
+struct Node {
+	int data;
+	Node* parent;
+	Node* left_child;
+	Node* right_child; 
+};
+class Tree {
+public:
+	Node* root;
+	Tree() {
+		root = nullptr;
```

Comment:
I think you can remove this constructor :)

When you declare `Node* root;` in the class definition, it's actually initialised to the default value at declaration time, so it's already `nullptr` before `Tree()` is even called.